src/main/java/seedu/duke/Duke.java

```diff
@@ -18,7 +18,7 @@ public class Duke {
     private static BudgetPlanner budgetPlanner = new BudgetPlanner();
     private static UI ui = new UI();
 
-    private static Parser parser = new Parser();
+    private static Parser parser = Parser.getInstance();
```

Comment:
I think can actually put to null instead of initialising here cause at Duke will initialise